tests/memory_comparator.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+
+import sys
+from collections import Counter
+
+def main():
+    filename1 = sys.argv[1]
+    filename2 = sys.argv[2]
+    cairo_mem = Counter()
+    cleo_mem = Counter()
+
+    with open(filename1, 'rb') as f:
+        byte = f.read(40)
+        cairo_mem.update(byte)
+
+    with open(filename2, 'rb') as f:
+        byte = f.read(40)
+        cleo_mem.update(byte)
```

Comment:
But there's no loop. It'll read the first k/v only. Besides, the byte array is an iterable, `cleo_mem.update` will count by byte rather than the block. You would need to construct a list of byte arrays.
Probably something like:
```python3
cleo_mem.update(byte for byte in f.read(40))
```
We should see how EOF behaves tho, it may be wiser to explictly loop.

Reply:
The current behavior is similar to this:
```python3
>>> from collections import Counter
>>> b = bytes([1]*40)
>>> c = Counter()
>>> c.update(b)
>>> c
Counter({1: 40})
>>> c
```

Reply:
https://github.com/lambdaclass/cleopatra_cairo/pull/247/commits/022a31ad7a43c233ed6e78c932d1e28b80dbd372